ursa/graph/utils.py

```diff
@@ -44,3 +47,64 @@ def _apply_append(collection, values):
             collection.update(val)
 
         return collection
+
+
+@ray.remote(num_return_vals=0)
+def write_vertex(vertex, graph_id, key):
+    dest_directory = graph_id + "/" + key + "/"
+
+    # @TODO(kunalgosar): Accessing _transaction_id here is not recommended
+    dest_file = dest_directory + str(vertex._transaction_id) + ".dat"
+    if not os.path.exists(dest_directory):
+        os.makedirs(dest_directory)
+
+    serialization_context = ray.worker.global_worker.serialization_context
+
+    oids = [vertex.vertex_data]
+    oids.extend(vertex.local_edges.edges)
+    local_edges_buffer = pa.serialize(vertex.local_edges.buf,
+                                      serialization_context).to_buffer()
+
+    oids.append(local_edges_buffer)
+    num_local_edge_lists = len(vertex.local_edges.edges) + 1
+
+    foreign_edges = list(vertex.foreign_edges.keys())
+    oids.extend([vertex.foreign_edges[k] for k in foreign_edges])
+    oids = [pa.plasma.ObjectID(oid.id()) for oid in oids
+            if type(oid) is ray.local_scheduler.ObjectID]
+
+    buffers = ray.worker.global_worker.plasma_client.get_buffers(oids)
+    data = {"vertex_data": buffers[0],
+            "local_edges": buffers[1:1 + num_local_edge_lists],
+            "foreign_edges": foreign_edges,
+            "foreign_edge_values": buffers[1 + num_local_edge_lists:],
+            "transaction_id": vertex._transaction_id}
+
+    serialized = pa.serialize(data, serialization_context).to_buffer()
+    with pa.OSFile(dest_file, 'wb') as f:
+        f.write(serialized)
+
+
+@ray.remote
+def read_vertex(file):
```

Comment:
This needs proper docstring commenting (see Slack message for reference)

Reply:
Resolved.